chain/network/src/types.rs

```diff
@@ -104,33 +109,56 @@ impl PeerInfo {
     }
 }
 
+// Note, `Display` automatically implements `ToString` which must be reciprocal to `FromStr`.
 impl fmt::Display for PeerInfo {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if let Some(acc) = self.account_id.as_ref() {
-            write!(f, "({}, {:?}, {})", self.id, self.addr, acc)
+        write!(f, "{}", self.id)?;
+        if let Some(addr) = &self.addr {
+            write!(f, "@{}", addr)?;
+        }
+        if let Some(account_id) = &self.account_id {
+            write!(f, "@{}", account_id)?;
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for PeerInfo {
+    type Err = Box<dyn std::error::Error>;
```

Comment:
let's avoid Box<error> where we can - it's really hard to operate with it. if this is std::io::Error really - let's return that

Reply:
We are mixing `std::io:Error` and `AddrParseError` here.